tests/s3/test_multipart_getput.py

```diff
@@ -758,3 +757,102 @@ def test_multipart_upload_test_28528(self):
+        self.create_file_mpu(mp_config["file_size"], self.mp_obj_path)
+        parts = get_precalculated_parts(self.mp_obj_path, mp_config["part_sizes"],
+                                        chunk_size=mp_config["chunk_size"])
+        keys = list(parts.keys())
+        random.shuffle(keys)
+        mpu_id = self.initiate_multipart(self.bucket_name, self.object_name)
+        self.log.info("Uploading parts")
+        status, new_parts = self.s3_mpu_test_obj.upload_parts_sequential(mpu_id, self.bucket_name,
+                                                                         self.object_name,
+                                                                         parts=parts)
+        assert_utils.assert_true(status, f"Failed to upload parts: {new_parts}")
+        self.log.info("Listing parts of multipart upload")
+        res = self.s3_mpu_test_obj.list_parts(mpu_id, self.bucket_name, self.object_name)
+        assert_utils.assert_true(res[0], res[1])
+        self.log.info("Listed parts of multipart upload: %s", res[1])
+        sorted_part_list = sorted(new_parts, key=lambda x: x['PartNumber'])
+        self.log.info("Complete the multipart upload")
+        try:
+            resp = self.s3_mpu_test_obj.complete_multipart_upload(
+                mpu_id, sorted_part_list, self.bucket_name, self.object_name)
+            assert_utils.assert_true(resp[0], resp[1])
+        except CTException as error:
+            self.log.error(error)
+            self.log.info("Failed to complete the multipart")
+        self.get_obj_compare_checksums(self.bucket_name, self.object_name, resp[1]["ETag"])
+        status, put_etag = self.s3_test_obj.put_object(self.bucket_name, self.object_name,
+                                                       self.mp_obj_path)
+        assert_utils.assert_true(status, put_etag)
+        self.log.info("Put object ETag: %s", put_etag)
+        self.compare_checksums(put_etag, resp[1]["ETag"])
+        self.log.info("Stop and validate parallel S3 IOs")
+        s3_background_io.stop()
+        s3_background_io.cleanup()
+        self.log.info("ENDED: Test Simple and Multipart upload of an object")
+
+    @pytest.mark.tags('TEST-40256')
+    @pytest.mark.s3_ops
+    @CTFailOn(error_handler)
+    def test_multipart_upload_test_40256(self):
+        """
+        This test is for head-object of multipart uploaded object
+        """
+        mp_config = MPART_CFG["test_28539"]
+        m_key = "mpukey"
+        s3_background_io = S3BackgroundIO(s3_test_lib_obj=self.s3_test_obj)
+        self.log.info("STARTED: test head-object of multipart uploaded object")
+        self.log.info("start s3 IO's")
+        s3_background_io.start(log_prefix="TEST-28539_s3bench_ios", duration="0h2m")
+        self.create_file_mpu(mp_config["file_size"], self.mp_obj_path)
+        parts = get_precalculated_parts(self.mp_obj_path, mp_config["part_sizes"],
```

Comment:
Same set of pattern is followed in all tests. Please extract a function at lib layer and reduce the lines in file.

Reply:
Sure. Done

Reply:
https://jts.seagate.com/browse/TEST-40372 Tests executed after code changes